pkg/attestlib/e2etest/pkix_e2e_test.go

```diff
@@ -0,0 +1,144 @@
+oujjRhvcxXGmqIt4OSjdQw==
+-----END PRIVATE KEY-----`
+
+const rsa2048PubKey = `-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA1Vf+g3iALHB5GHz2sTMm
+VmemVMio8iAgIXDhfdJmAJJCUG61vH7ZFOJNWwVPiX2HE4iC8quLT3a2W5h81OiD
+Y2FXfD5vZB0lycXNZoasyhUlC4TFsL01tgh7W7WN5iBDlwxY13bcgv79SIbroz/C
++kS1+cqu4GQXmEHLYFg80pQVe7ssBaQ3qxA0HL0heXJfM0Ye40Aw3aC430h92f2a
+5JgY8JEqRTtYgh3VVuqzm3L4QvSWiHzfB29BXCB0GstFK448aYkk2RPI6Q1LkoT7
+NiCquPVF1EYnUXKrC+ANoWr/l7ldEJ7V4vpo+9EClnXcxAzq/knqeN5WdM6iPYny
+qwIDAQAB
+-----END PUBLIC KEY-----`
+
+const rsa3072PubKey = `-----BEGIN PUBLIC KEY-----
+MIIBojANBgkqhkiG9w0BAQEFAAOCAY8AMIIBigKCAYEA174cl1EgPn1w/4r1OTNy
+O5ViVEtjPV1Bl2xXA/4EOcVyvztdwovJriC2sry+gW7WS/YraS6BXTY328daY7u5
+OYAHnr/NiF3BoAzbAYMpHU6otMiAlGADrgevESj5XKUp4+XuiYxdOPH7pqgHaZ0+
+ZIvp08vH+xXsY1+WqA3TBsKxkCwTmWGUSy/j2Gml6XHgX1SywJGJ9VEI/5wACExX
+3NYHx+OXBS5XyjFzIoBxFiuA4g8FRHcTw1uulX98Rt/WTwTjesMeWaB7mOxgK03A
+rTOil11+/Wgboow3V93B5pdNMvVjkahYHvkjYkj/HJRuXUKidZANhBWEewpdHDkK
++NXa8BlxcbkzcO2DvXGRu1g0emVInoIEN+/1fz4ab/mTWUGgs9QX+INChlSV3EBz
+5I6mXbc3fE4EW2W9KASIPKIpF1HBaF58B//W+XAy7mxhuSfg23/mkf4SI9ZVM82s
+rBmJaTmRgKWbZXNaf4yH+RhyGF1CsBN2uLrt667Swxv3AgMBAAE=
+-----END PUBLIC KEY-----`
+
+func TestPkixEndToEnd(t *testing.T) {
+	tcs := []struct {
+		name                string
+		privateKey          []byte
+		publicKey           []byte
+		signatureAlgorithm  attestlib.SignatureAlgorithm
+		payload             []byte
+		image               string
+		expectedCreateError bool
+		expectedVerifyError bool
+	}{
+		{
+			name:                "verify attestation success",
+			privateKey:          []byte(rsa2048PrivateKey),
+			publicKey:           []byte(rsa2048PubKey),
+			signatureAlgorithm:  attestlib.RsaSignPkcs12048Sha256,
+			payload:             []byte(e2eTestPayload),
+			image:               e2eTestImage,
+			expectedCreateError: false,
+			expectedVerifyError: false,
+		}, {
+			name:                "verify failed wrong public key",
+			privateKey:          []byte(rsa2048PrivateKey),
+			publicKey:           []byte(rsa3072PubKey),
+			signatureAlgorithm:  attestlib.RsaSignPkcs12048Sha256,
+			payload:             []byte(e2eTestPayload),
+			image:               e2eTestImage,
+			expectedCreateError: false,
+			expectedVerifyError: true,
+		}, {
+			name:                "verify failed image does not match payload",
+			privateKey:          []byte(rsa2048PrivateKey),
+			publicKey:           []byte(rsa2048PubKey),
+			signatureAlgorithm:  attestlib.RsaSignPkcs12048Sha256,
+			payload:             []byte(e2eTestPayload),
+			image:               wrongTestImage,
+			expectedCreateError: false,
+			expectedVerifyError: true,
+		},
+	}
+	for _, tc := range tcs {
+		signer, err := attestlib.NewPkixSigner(tc.privateKey, tc.signatureAlgorithm, "")
+		if err != nil {
+			t.Fatalf("Error initializing signer: %v", err)
+		}
+		att, err := signer.CreateAttestation(tc.payload)
+		if tc.expectedCreateError {
```

Comment:
this won't work because if you are expecting create error and get one, it will still try to run the rest of the test code. I would make a separate test for create errors rather then putting everything below this in an else block.

Reply:
we probably won't need to test creationError, as it can be covered in signer unit test. (although the same goes for the existing pgp e2e tests, which necessarily has `wantCreateErr` always set to `false`)

Reply:
done.